mlkem/ntt.c

```diff
@@ -180,8 +180,11 @@ void basemul(int16_t r[2], const int16_t a[2], const int16_t b[2],
  **************************************************/
 void basemul_cached(int16_t r[2], const int16_t a[2], const int16_t b[2], int16_t b_cached)
 {
-    r[0] = fqmul(a[1], b_cached);
-    r[0] += fqmul(a[0], b[0]);
-    r[1] = fqmul(a[0], b[1]);
-    r[1] += fqmul(a[1], b[0]);
+    int32_t t0, t1;
+    t0  = (int32_t) a[1] * b_cached;
+    t0 += (int32_t) a[0] * b[0];
```

Comment:
Please justify (informally for now) why these * and + operators do not overflow int16_t. What do we require the ranges of a[i], b[i] and b_cached to be? Please document those things as a precondition (in a comment) for now. CBMC proof of this to follow!

Reply:
This may overflow, that's why it's computed in `int32_t` -- it's Montgomery reduced at the end.

Reply:
if a[0] == b[0] == a[1] == b_cached == -32768
then
 a[1] * b_cached + a[0] * b[0] == 2*31
which does not fit in int32_t, so UB.

Gissa comment that says this can't happen.

Reply:
Ah, ok. Here the `a` and `b` inputs are in `-Q,...,+Q` I _think_, or even signed canonical. Undoubtedly this needs working out as part of #43.